x/lscosmos/keeper/msg_server.go

```diff
@@ -541,3 +541,51 @@ func (m msgServer) JumpStart(goCtx context.Context, msg *types.MsgJumpStart) (*t
 
 	return &types.MsgJumpStartResponse{}, nil
 }
+
+// RecreateICA defines a method for recreating closed ica channels
+func (m msgServer) RecreateICA(goCtx context.Context, msg *types.MsgRecreateICA) (*types.MsgRecreateICAResponse, error) {
+	ctx := sdktypes.UnwrapSDKContext(goCtx)
+	// sanity check for the arguments of message
+	if ctx.IsZero() {
```

Comment:
hmm is this correct? `context.Context` we check if `isZero`

Reply:
i think it is not required now, because it is added in abci, wonder from where it arrived first.